### PR TITLE
Reorder user stories and update NFR and glossary

### PR DESCRIPTION
**Pull request by kangqiao322**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -621,52 +621,37 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 2. Should be able to hold up to 1000 internship applications without a noticeable sluggishness in performance for typical usage.
 3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-4. Data should be stored consistently upon closing and reopening the app
-5. System should be usable by a Computing student who has never applied to an internship before
-6. System should be easily usable by the majority of Year 2+ Computing students
-
-
-*{More to be added}*
-
+4. Data should be stored consistently upon closing and reopening the app.
```
Comment (jetlfj) [inline docs/DeveloperGuide.md:637]: Data is stored when it is changed though?

Comment (kangqiao322) [reply docs/DeveloperGuide.md:637]: what do you mean?

Comment (jetlfj) [reply docs/DeveloperGuide.md:950]: Data is not stored when the app is closed or opened. As stated in the user guide, "data is saved in the hard disk automatically after any command that changes the data."

Comment (kangqiao322) [reply docs/DeveloperGuide.md]: I have updated the wording.

**Comment by chantellyu** [comment]
LGTM!

**Pull request merged**